src/test/java/rx/internal/operators/OperatorObserveOnTest.java

```diff
@@ -834,4 +834,16 @@ public void testErrorDelayedAsync() {
         ts.assertError(TestException.class);
         ts.assertNotCompleted();
     }
+    
+    @Test
+    public void requestExactCompletesImmediately() {
```

Comment:
Not sure if it tests "immediately", this test passes without changes in this PR.

Maybe you can avoid going to another thread and `awaitTerminalEvent()`?